routing/road_access.hpp

```diff
@@ -1,32 +1,77 @@
 #pragma once
 
+#include "routing/segment.hpp"
+#include "routing/vehicle_mask.hpp"
+
+#include "base/assert.hpp"
+
 #include <cstdint>
+#include <map>
 #include <string>
 #include <vector>
 
 namespace routing
 {
 // This class provides information about road access classes.
-// For now, only restrictive types (such as barrier=gate and access=private)
-// and only car routing are supported.
+// One instance of RoadAccess holds information about one
+// mwm and one router type (also known as VehicleType).
 class RoadAccess final
 {
 public:
-  std::vector<uint32_t> const & GetPrivateRoads() const { return m_privateRoads; }
+  // The road access types are selected by analyzing the most
+  // popular tags used when mapping roads in OSM.
+  enum class Type : uint8_t
+  {
+    // Moving through the road is prohibited.
+    No,
+
+    // Moving through the road requires a special permission.
+    Private,
+
+    // No transit through the road is allowed; however, it can
+    // be used if it is close enough to the destination point
+    // of the route.
+    Destination,
+
+    // No restrictions, as in "access=yes".
+    Yes,
+
+    // The number of different road types.
+    Count
+  };
+
+  RoadAccess();
+
+  RoadAccess(VehicleType vehicleType);
```

Comment:
Why do you need m_vehicleType? You don't use it in the class, and m_vehicleType is not even used in comparison operator.